### ISSUE TITLE
Error recording from www.iranntv.com/?p=281476

### ISSUE THREAD

**Issue by hejazb** [NapoleonWils0n/kodi-playercorefactory#8]
Hi Dear: Napoleon
I was finally able to find a way to watch "www.iranntv.com/?p=281476" stream in Kodi 15.2 but when I try to record while in Kodi or using Bash in windows shell, I get the following errors. I updated my Playercorefactory folder but it did not help could you look at the errors and see if you can fine a cure.
Thank you so very much.
Best wishes
Harry
# EXTM3U
# EXTINF:-1,**Simay Azadi IRAN***

http://rrr.sz.xlcdn.com/?account=simay&file=MultiL&type=live&service=wowza&protocol=http&output=playlist.m3u8
Recording from Kodi Simple IPTV PVR

Error
"oaded to different address: parent(0x370000) != child(0x3A0000)
/cygdrive/c/Users/Boby/AppData/Roaming/Kodi/userdata/kodi-playercorefactory/bash
-scripts-windows/video-regex.sh: fork: retry: No child processes
      2 [main] bash 3172 child_info_fork::abort: C:\cygwin\bin\cyggcc_s-1.dll: L
oaded to different address: parent(0x370000) != child(0x3A0000)
/cygdrive/c/Users/Boby/AppData/Roaming/Kodi/userdata/kodi-playercorefactory/bash
-scripts-windows/video-regex.sh: fork: retry: No child processes"

Recording using Bash rip-record
C:\Users\Boby\Desktop>bash rip-record videourl.txt
Error
"pvr://channels/tv/All channels/pvr.iptvsimple_526342641.pvr  : Protocol not foun
d"

**Comment by NapoleonWils0n** [comment]
Hi Harry
I have updated the scripts and you can now record, play and stream irantv within Kodi and on the command line
I just had to add the & symbol to the regular expression for m3u8 video streams

Do a git pull to update

But you cant record from simple iptv pvr, you need to use playlist loader or create a .strm file instead i talk you through the steps, i explain the problem with simple iptv pvr at the end of this comment

I have created a plx playlist for iranntv for playlist loader and a strm and added them to the git repository

https://github.com/NapoleonWils0n/kodi-playercorefactory/tree/master/playlists/plx-playlists
https://github.com/NapoleonWils0n/kodi-playercorefactory/tree/master/playlists/strm

Playlist loader

1 - Download and install xbmc Israel Repo and then install playlist loader:

https://seo-michael.co.uk/tutorial-how-to-install-playlist-loader-kodi-xbmc/

heres the xbmc Israel zip file:

https://github.com/cubicle-vdo/xbmc-israel/blob/master/repo/repository.xbmc-israel/repository.xbmc-israel-1.0.4.zip?raw=true

2 -  add the iranntv.plx playlist from the git repo to playlist loader

if you want to change the playlist file to change the url or add more links heres what to do

copy the iranntv.plx file from the kodiplaycore-factory/playlist/iranntv.plx  to another folder on your computer like Video/playlists so that if the url changes you can edit it, otherwise if you make changes to the iranntv.plx in the git repo and then do a git pull it would overwrite your changes

you can create either a local playlist where you copy the playlist file to the machine running kodi, or a remote playlist where you add in a url

local playlist

open the playlist loader addon
click on Add a new list
name the list iranntv
in the next window select playlist source: select Local-list (file)
browse to where you copied the iranntv.plx file and select it
then it asks you to choose a list logo: select none

remote playlist

open the playlist loader addon
click on Add a new list
name the list iranntv
in the next window select playlist source: select Remote-list (file)
type in this url: 
https://raw.githubusercontent.com/NapoleonWils0n/kodi-playercorefactory/master/playlists/plx-playlists/iranntv.plx
then it asks you to choose a list logo: select none

that will add the playlist to kodi
i have added the iranian flag and channel logo to the playlist as well

STRM files

you can also create a text file with a .strm file extension and add the video url 
like this

https://github.com/NapoleonWils0n/kodi-playercorefactory/blob/master/playlists/strm/iranntv.strm

Then put the iranntv.strm on a video share you can access from kodi
then just click on the iranntv.strm in kodi and it will open the video

I have tested recording and playing iranntv on the command line and from within Kodi using the playlist in playlist loader and the .strm file and everything works perfectly

If you need any help with anything leave a comment and ill get back to you

simple iptv pvr problem and why you cant record

If you look at error message when you tried to record from the simple iptv pvr 
"pvr://channels/tv/All channels/pvr.iptvsimple_526342641.pvr : Protocol not foun
d"
Basically what happens is when you try and save the video url from the within the simple iptv pvr it creates a url for the pvr channel and not the actual url for the video

so in the simple iptv pvr client it creates this url:
pvr://channels/tv/All channels/pvr.iptvsimple_526342641.pvr

but the actually url is:
http://rrr.sz.xlcdn.com/?account=simay&file=MultiL&type=live&service=wowza&protocol=http&output=playlist.m3u8

So you cant record from within simple iptv pvr and need to use playlist loader or a .strm file

All the best

**Comment by hejazb** [comment]
Hi Dear: Napoleon
Per your instruction now, I can record IRANNTV, thank you thank you.
Do you have positive experience recording Kodi stream with Android TV Box (5.1 or 4.4)? I think you said: you only worked in virtual Linux machines. I like to spend time on learning to build an Android TV box that can record like a windows box that you thought me to build, I just want to know your experience or others that might have done this before and if there is a simple tutorial like yours that I can follow. I surely appreciate it. Kodi Android WIKI says at present there is no support for Kodi recording.
Thank you so much for your wonderful job and helping others with your support and let them enjoy using your great achievement to advance their knowledge.
I wish you best
Harry
